crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -582,20 +573,25 @@ impl LogletConfiguration {
     fn try_reconfiguring(
         &self,
         log_id: LogId,
+        last_segment_index: SegmentIndex,
         logs_configuration: &LogsConfiguration,
         observed_cluster_state: &ObservedClusterState,
         preferred_sequencer: Option<NodeId>,
     ) -> Option<LogletConfiguration> {
-        let loglet_id = self.loglet_id();
+        let next_loglet_id = match self {
+            Self::Local(_) => LogletId::new(log_id, last_segment_index.next()),
+            #[cfg(any(test, feature = "memory-loglet"))]
+            Self::Memory(_) => LogletId::new(log_id, last_segment_index.next()),
```

Comment:
Maybe add a comment why it is ok to ignore the previous loglet id of `Local` and `Memory` and set it to `LogletId::new(log_id, last_segment_index.next())`.

Reply:
I like this approach, it's practically safe as we'd only do reconfiguration of those in tests only.